embassy-stm32/src/usart/buffered.rs

```diff
@@ -396,6 +396,13 @@ impl<'d> BufferedUart<'d> {
         (self.tx, self.rx)
     }
 
+    /// Split the Uart into a transmitter and receiver by mutable reference,
+    /// which is particularly useful when having two tasks correlating to
+    /// transmitting and receiving.
+    pub fn split_ref(&mut self) -> (&mut BufferedUartTx<'d>, &mut BufferedUartRx<'d>) {
```

Comment:
Hi Dirbaio 

I'm not 100 % sure how to implement this. I guess i would need to impl copy for BufferedUartTx and RX?
Should I also change the implementation for rp and nrf as I copied the function split_ref from there?

Reply:
> impl copy for BufferedUartTx and RX

that would allow the user to copy it themselves, which we don't want.

You can get copies (with shorter lifetime) of `PeripheralRef`s with `.reborrow()`. I think it should be possible to do that in this case, but as an unsafe escape hatch there's also `.clone_unchecked()`.